src/configureWorkspace/scaffolding.ts

```diff
@@ -138,14 +141,32 @@ export async function scaffold(context: ScaffoldContext): Promise<ScaffoldedFile
     await Promise.all(
         files.map(
             async file => {
-                const filePath = path.resolve(rootFolder, file.fileName);
-
-                if (await fse.pathExists(filePath) === false || await promptForOverwrite(file.fileName)) {
-                    await fse.writeFile(filePath, file.contents, 'utf8');
-
-                    writtenFiles.push({ filePath, open: file.open });
+                let filePath = path.resolve(rootFolder, file.fileName);
+
+                if (await fse.pathExists(filePath)) {
+                    if (file.createNewNameIfExist) {
+                        filePath = await generateNonConflictFileName(filePath);
+                    } else if (await promptForOverwrite(file.fileName) === false) {
+                        return;
+                    }
                 }
+
+                await fse.writeFile(filePath, file.contents, 'utf8');
+                writtenFiles.push({ filePath, open: file.open });
             }));
 
     return writtenFiles;
+
+    async function generateNonConflictFileName(filePath: string): Promise<string> {
+        let newFilepath = filePath;
+        let i = 1;
+        const extName = path.extname(filePath);
+        const extNameRegEx = new RegExp(`${extName}$`);
+
+        while (await fse.pathExists(newFilepath)) {
+            newFilepath = filePath.replace(extNameRegEx, i + extName);
```

Comment:
It might be better to split apart the base file name from the directory and extension, and only update the base file name.  This regex has a risk of replacing a pattern elsewhere in the path.  For example, imagine a filename like: `/repo/config.yamls/docker-compose.yaml`.

Reply:
The $ looks for ext name at the end. So it is safe.

Reply:
Yes, sorry, missed that trailing `$`.